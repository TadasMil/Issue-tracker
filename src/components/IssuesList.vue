<script setup lang="ts" name="IssuesList">
import type { IIssue } from '@/types'

import IssuesListRow from './IssuesListRow.vue'
import IssuesListEmpty from './IssuesListEmpty.vue'

const { issues } = defineProps<{
  issues: IIssue[]
}>()

const emit = defineEmits<{
  (event: 'edit', issue: IIssue): void
  (event: 'mark-as-trashed', issue: IIssue): void
  (event: 'mark-as-completed', issue: IIssue): void
}>()
</script>

<template>
  <div v-if="issues.length">
    <IssuesListRow
      v-for="issue in issues"
      :key="issue.id"
      :title="issue.title"
      :description="issue.description"
      @edit="emit('edit', issue)"
      @mark-as-trashed="emit('mark-as-trashed', issue)"
      @mark-as-completed="emit('mark-as-completed', issue)"
    />
  </div>

  <template v-else>
    <IssuesListEmpty />
  </template>
</template>
