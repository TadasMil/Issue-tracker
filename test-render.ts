import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
