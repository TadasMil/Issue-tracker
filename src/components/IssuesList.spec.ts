import { describe, it, expect } from 'vitest'
import { render } from '../../test-render'

import IssuesList from './IssuesList.vue'

describe('IssuesList', () => {
  it('should render the component', () => {
    const { html } = render(IssuesList, {
      props: {
        issues: [
          {
            id: '1',
            title: 'My first issue',
            description: 'My first issue description',
            status: 'open',
          },
        ],
      },
      global: {
        stubs: ['IssuesListRow'],
      },
    })

    expect(html()).toMatchSnapshot()
  })

  it('should render empty component state', () => {
    const { html } = render(IssuesList, {
      props: {
        issues: [],
      },
      global: {
        stubs: ['IssuesListEmpty'],
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
