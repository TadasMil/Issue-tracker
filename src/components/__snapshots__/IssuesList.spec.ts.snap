// Vitest Snapshot v1

exports[`IssuesList > should render empty component state 1`] = `"<issues-list-empty-stub></issues-list-empty-stub>"`;

exports[`IssuesList > should render the component 1`] = `
"<div>
  <issues-list-row-stub title=\\"My first issue\\" description=\\"My first issue description\\"></issues-list-row-stub>
</div>"
`;
